controllers/openlibertyapplication_controller.go

```diff
@@ -489,12 +520,32 @@ func (r *ReconcileOpenLiberty) Reconcile(ctx context.Context, request ctrl.Reque
 				}
 			}
 
+			if r.isPasswordEncryptionKeySharingEnabled(instance) && len(encryptionSecretName) > 0 {
+				lutils.ConfigurePasswordEncryption(&statefulSet.Spec.Template, instance, OperatorShortName)
+				lastRotationAnnotation, err := lutils.GetSecretLastRotationAsLabelMap(instance, r.GetClient(), encryptionSecretName, PASSWORD_ENCRYPTION_RESOURCE_SHARING_FILE_NAME)
+				if err != nil {
+					return err
+				}
+				instance.Annotations = oputils.MergeMaps(instance.Annotations, lastRotationAnnotation)
+				if instance.Status.GetReferences()[lutils.GetTrackedResourceName(PASSWORD_ENCRYPTION_RESOURCE_SHARING_FILE_NAME)] != encryptionSecretName {
+					instance.Status.SetReference(lutils.GetTrackedResourceName(PASSWORD_ENCRYPTION_RESOURCE_SHARING_FILE_NAME), encryptionSecretName)
+					updateStatus = true
+				}
+				updateAnnotations = true
+			}
+
 			if r.isLTPAKeySharingEnabled(instance) && len(ltpaSecretName) > 0 {
-				lutils.ConfigureLTPA(&statefulSet.Spec.Template, instance, OperatorShortName)
-				err := lutils.AddSecretResourceVersionAsEnvVar(&statefulSet.Spec.Template, instance, r.GetClient(), ltpaSecretName, "LTPA")
+				lutils.ConfigureLTPA(&statefulSet.Spec.Template, instance, OperatorShortName, ltpaSecretName, ltpaMetadata.Name)
+				lastRotationAnnotation, err := lutils.GetSecretLastRotationAsLabelMap(instance, r.GetClient(), ltpaSecretName, LTPA_RESOURCE_SHARING_FILE_NAME)
 				if err != nil {
 					return err
 				}
+				instance.Annotations = oputils.MergeMaps(instance.Annotations, lastRotationAnnotation)
```

Comment:
Annotation on CR instance will apply to all k8s resources managed by the Operator and polluting them, also it won't be removed once added if/when LTPA/password encryption is later disabled. 

The last rotation annotation should instead directly be added on PodTemplateSpec of Deployment/StatefulSet, similar to lutils.AddSecretResourceVersionAsEnvVar method.